src/app_service/service/provider/service.nim

```diff
@@ -289,11 +289,11 @@ proc process(self: Service, data: APIRequest): string =
     "data": value
   }
 
-method postMessage*(self: Service, message: string): string =
-  case message.requestType():
-  of RequestTypes.Web3SendAsyncReadOnly: self.process(message.toWeb3SendAsyncReadOnly())
-  of RequestTypes.HistoryStateChanged: """{"type":"TODO-IMPLEMENT-THIS"}""" ############# TODO:
-  of RequestTypes.APIRequest: self.process(message.toAPIRequest())
+method postMessage*(self: Service, requestType: string, message: string): string =
```

Comment:
@jrainville isn't easier to have here `RequestTypes` as a type of `requestType` param, than `string?`

Reply:
Updated